File: 2022/scala/src/test/scala/AdventOfCode2022/Day06Test.scala
package AdventOfCode2022

import org.scalatest.flatspec.AnyFlatSpec

class Day06Test extends AnyFlatSpec {
  val example = "mjqjpqmgbljsphdztnvjfqwrcgsmlb"

  "Part one" should "provide the correct answer for example input" in {
    assert(Day06.partOne(example) == 7)
  }

  "Part two" should "provide the correct answer for example input" in {
    assert(Day06.partTwo(example) == 19)
  }
}

File: 2024/python/day06_test.py
"""
Advent of Code 2024, Day 6: Guard Gallivant.

See: https://adventofcode.com/2024/day/6
"""

import io

import pytest

from day06 import part_one, part_two

EXAMPLE = """\
....#.....
.........#
..........
..#.......
.......#..
..........
.#..^.....
........#.
#.........
......#...
"""


@pytest.fixture()
def example() -> str:
    return EXAMPLE


def test_part_one(example: str) -> None:
    """
    Test that checks if the solution for part one works on the example input.
    """
    assert part_one(io.StringIO(example)) == 41


def test_part_two(example: str) -> None:
    """
    Test that checks if the solution for part two works on the example input.
    """
    assert part_two(io.StringIO(example)) == 6
